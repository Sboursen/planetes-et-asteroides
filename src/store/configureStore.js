import {
  applyMiddleware,
  createStore,
  combineReducers,
} from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import getBodiesReducer from './home/Home';
import getBodyReducer from './details/Details';

const rootReducer = combineReducers({
  bodies: getBodiesReducer,
  body: getBodyReducer,
});

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
