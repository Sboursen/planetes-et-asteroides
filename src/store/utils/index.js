const defaultPriority = {
  Planet: 1,
  'Dwarf Planet': 2,
  Moon: 3,
  Star: 4,
  Asteroid: 5,
  Comet: 6,

};

export default function orderByPriority(
  bodies,
  property,
  priority = defaultPriority,
) {
  return bodies.sort(
    (bodyA, bodyB) => priority[bodyA[property]] - priority[bodyB[property]],
  );
}
