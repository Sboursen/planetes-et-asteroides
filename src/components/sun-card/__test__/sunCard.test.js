import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import store from '../../../store/configureStore';
import SunCard from '../sunCard';

describe('SunCard', () => {
  afterEach(cleanup);

  it('renders correctly in the DOM', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <SunCard />
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the exact name', () => {
    render(
      <Provider store={store}>
        <Router>
          <SunCard />
        </Router>
      </Provider>,
    );

    const nameElement = screen.getByText('The Sun');
    const { classList } = nameElement;

    expect(classList).toContain('whitespace-nowrap');
    expect(classList).toContain('font-control');
  });
});
