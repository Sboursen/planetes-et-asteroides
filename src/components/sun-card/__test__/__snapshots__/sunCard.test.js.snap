// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SunCard renders correctly in the DOM 1`] = `
<div
  className="flex flex-row justify-around items-start shadow-sm rounded py-2"
>
  <div
    className="flex flex-col w-fit items-start"
  >
    <h2
      className="text-5xl font-control whitespace-nowrap"
    >
      The Sun
    </h2>
    <ul
      className="flex flex-col pl-4"
    />
  </div>
  <ul
    className="border-2 border-white bg-grid-one rounded p-3"
  >
    <li>
      Type: Star
    </li>
    <li
      className="block"
    >
      <span
        className="inline"
      >
        Mass: 
      </span>
      <pre
        className="inline text-base"
      >
        1.989
        x 10
        <sup>
          30
        </sup>
        kg
      </pre>
    </li>
    <li
      className="block"
    >
      <span
        className="inline"
      >
        Volume: 
      </span>
      <pre
        className="inline text-base"
      >
        1.409
         
        x 10
        <sup>
          18
        </sup>
        km
        <sup>
          3
        </sup>
      </pre>
    </li>
  </ul>
</div>
`;
